### PR TITLE
643 fire cli for manual remove

### LINKED ISSUE #643: Fire CLI for manual remove script
Issue raised by: [developer]

### What?
After the successful changes to manual_submission to make it use fire CLI #637 we want to do the same for the manual_remove script. Similar to manual submission we want to replace the argparse information and handling with fire CLI. See the linked pr #637 for details

### Where?
* unit tests: `scripts/manual_operations/tests/test_manual_remove.py`
* source code: `scripts/manual_operations/tests/test_manual_submission.py`

### How?
Continuation of using new CLI tool

### How to test the issue is resolved
* Use the manual_remove script (following the README.md instructions `scripts/manual_operations/README.md`)
* Ensure these functions still pass
* Ensure all travis unit tests pass and are updated where required

### PR DESCRIPTION
**Pull request by keiranjprice101**
### Summary of work

Used #637 as a guide. Not sure if I should be adding any labels, so please let me know :)

I added mocking for `user_input_check` as without it the test would hang until y/n was entered. (I don't understand why this wasn't happening previously)

I didn't add any type casting as from the docs https://google.github.io/python-fire/guide/#argument-parsing this should be automatic.

### How to test your work
 
The only test failing is `test_find_run` though this was failing already due to my sub optimal setup.
### Additional comments
 

closes #643 

**Before merging ensure the release notes have been updated**

### CODE REVIEW

File: scripts/manual_operations/tests/test_manual_remove.py [hunk 1]
```diff
@@ -208,16 +205,14 @@ def test_main(self, mock_delete, mock_process, mock_find):
     @patch('scripts.manual_operations.manual_remove.ManualRemove.find_runs_in_database')
     @patch('scripts.manual_operations.manual_remove.ManualRemove.process_results')
     @patch('scripts.manual_operations.manual_remove.ManualRemove.delete_records')
-    @patch('scripts.manual_operations.manual_remove.handle_input')
-    def test_main_range(self, mock_handle_input, mock_delete, mock_process, mock_find):
+    @patch('scripts.manual_operations.manual_remove.user_input_check')
+    def test_main_range(self, mock_delete, mock_process, mock_find, mock_uic):
```
Comment (JackEAllen) [inline scripts/manual_operations/tests/test_manual_remove.py:209]: It's a bit of a subtlety of python here where the patched method parameters are read in as arguments from bottom to top so mock_uic should be placed just after `self` such as `def test_main_range(self, mock_uic, mock_delete, mock_process, mock_find):`
Right now, mock_uic is representing `@patch('scripts.manual_operations.manual_remove.ManualRemove.find_runs_in_database')`

Comment (JackEAllen) [inline scripts/manual_operations/tests/test_manual_remove.py:209]: Could we have 2 more tests here? 
* Test the removal of an individual run and also a range of runs less than 10.
* Test invalid input - so maybe something to asset that an error is raised if the first run > end run

File: scripts/manual_operations/tests/test_manual_remove.py [hunk 2]
```diff
@@ -231,20 +231,6 @@ def test_main_range_less_than_ten(self, mock_delete, mock_process, mock_find):
         mock_process.assert_called()
         mock_delete.assert_called()
 
-    # pylint:disable=no-self-use
-    @patch('scripts.manual_operations.manual_remove.ManualRemove.find_runs_in_database')
-    @patch('scripts.manual_operations.manual_remove.ManualRemove.process_results')
-    @patch('scripts.manual_operations.manual_remove.ManualRemove.delete_records')
-    def test_main_single_run(self, mock_delete, mock_process, mock_find):
```
Comment (keiranjprice101) [inline scripts/manual_operations/tests/test_manual_remove.py]: This was just the exact same as test_main 🤦

File: scripts/manual_operations/manual_remove.py [hunk 3]
```diff
@@ -242,11 +213,22 @@ def user_input_check(instrument, run_numbers):
     return user_input
 
 
-def main():
+def main(instrument: str, first_run: int, last_run: int = None):
```
Comment (JackEAllen) [inline scripts/manual_operations/manual_remove.py:216]: This is probably my bad providing a poor description XD What i mean by adding the types is that we add the types in the docstring themselves so something like this:
`  :param instrument: (str) Instrument to run on
    :param first_run: (str) First run to be removed
    :param last_run: (str) Optional last run to be removed`

Comment (JackEAllen) [reply scripts/manual_operations/manual_remove.py:216]: As soon as this is changed, you can merge in 👍

Comment (keiranjprice101) [reply scripts/manual_operations/manual_remove.py:216]: My mistake 👍

**Comment by keiranjprice101** [comment]
@JackEAllen I notice you changed the base from develop to master. Are we merging straight into master now?

**Comment by JackEAllen** [comment]
Yes, as of yesterday 🙂

**Pull request merged**